snapcraft/cli/_options.py

```diff
@@ -121,6 +133,29 @@ def get_build_environment_flags(provider: str, **kwargs) -> Dict[str, str]:
     """Filter environment flags to relevant provider ones."""
     provider_env: Dict[str, str] = dict()
 
+    for key in ["http_proxy", "https_proxy"]:
+        if key not in kwargs:
+            continue
+
+        value = kwargs[key]
+        if not value:
+            continue
+
+        if provider in ["host", "managed-host", "lxd", "multipass"]:
+            provider_env[key] = value
+        elif not os.getenv(key):
```

Comment:
if this is the case, why not make the http-proxy and https-proxy params hidden?

Reply:
They can certainly be hidden.  Though I'm not sure I understand the purpose of making it harder to find valid options though...

I would consider the cases of hiding options to be good when feature is very experimental or deprecated.

In the case of http_proxy and https_proxy, I don't know that it is either.

If the concern is that it may flooding the user with advanced, unlikely to use options, I would suggest either:
(1) introduce a new "provider-specific options:" list below the standard options in `help`, or
(2) effectively group them by prefacing the names with `provider`, e.g. `--provider-http-proxy`.

Reply:
this comment is outdated from reading the code top-down and then bottom-up. It might be easier to grasp the objective if instead of a big loop some list comprehension is done to reduce the sets of what is trying to be accomplished and assigning to nice variable names.

Alternatively, more code comments can help.